simaple/simulate/component/mob.py

```diff
@@ -0,0 +1,83 @@
+import pydantic
+
+from simaple.simulate.base import Entity, Event
+from simaple.simulate.component.base import Component, ReducerState, reducer_method
+from simaple.simulate.reserved_names import Tag
+
+
+class DOT(Entity):
+    current: dict[str, tuple[float, float]] = pydantic.Field(
+        default_factory=dict
+    )  # name, damage, lasting_time
+    period_time_left: float = 1_000
+    period: int = 1_000
+
+    def new(self, name: str, damage: float, lasting_time: float) -> None:
+        self.current[name] = (damage, lasting_time)
+
+    def elapse(self, time: float) -> dict[tuple[str, float], int]:
+        emits: dict[tuple[str, float], int] = {}  # (name, damage): count
+
+        elapse_time_left = time
+        while elapse_time_left > 0:
+            elapse_time_left, events = self.step(elapse_time_left)
+            for k in events:
+                emits[k] = emits.get(k, 0) + 1
+
+        return emits
+
+    def step(self, time: float) -> tuple[float, list[tuple[str, float]]]:
+        if self.period_time_left > time:
+            self.period_time_left -= time
+            return 0, []
+
+        left_time = time - self.period_time_left
+        lapse_time = self.period_time_left
+
+        new_current = {}
+        events = []
+
+        for name, (damage, lasting_time) in self.current.items():
+            lasting_time -= lapse_time
+            if lasting_time >= 0:
+                new_current[name] = (damage, lasting_time)
+                events.append((name, damage))
```

Comment:
```suggestion
        new_current = {
          name: (damage, lasting_time - lapse_time)
          for name, (damage, lasting_time)
          in self.current.items()
          if lasting_time - lapse_time >= 0
        }
        events = [(name, damage) for name, (damage, _) in new_current.items()]
```
suggestion(non-blocking): 선언적으로 작성하는건 어떤가요?

Reply:
남은 시간이 있는 경우에 이벤트를 발생시킨다라는 점이 잘 안드러나는 것 같아서 주석 한줄만 추가해서 반영할게요.